src/middleware/context.go

```diff
@@ -0,0 +1,56 @@
+package middleware
+
+import (
+	"github.com/dana-team/platform-backend/src/customerrors"
+	"github.com/gin-gonic/gin"
+	"go.uber.org/zap"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// GetKubeClient retrieves the Kubernetes client from the gin.Context.
+func GetKubeClient(c *gin.Context) (kubernetes.Interface, error) {
+	kube, exists := c.Get(KubeClientCtxKey)
```

Comment:
dont see why its in middleware if we using it primarily in the routes  i think it could be in the routes

Reply:
except the last function AddErrorToContext

Reply:
These clients and the logger are set in the context by middleware, so it makes sense to have the retrieval functions in the middleware as well. 
`GetLogger`, `GetDynClient`, and `AddErrorToContext` are used within middlewares, so keeping them together in the middleware package maintains logical grouping.